packages/rspack-cli/src/rspack-cli.ts

```diff
@@ -76,6 +77,13 @@ export class RspackCLI {
 		};
 	}
 	async run(argv: string[]) {
+		if (semver.lte(semver.clean(process.version), "12.0.0")) {
+			this.getLogger().error(
+				`Node.js version must be greater than 12.0.0, current version is ${process.version}`
+			);
+			process.exit(1);
```

Comment:
Maybe it's better not to force an exit. I had the same thought as you in the first place, But I think a warning would be enough now. We don't promise Rspack could be running in node12 but if it could, there's no reason to forbid it.